### PR TITLE
feat(ssr): forward router instance to findResultsState clone

### PR DESCRIPTION
**Pull request by Haroenv**
in our guide for routing with vue router, you need to access the Vue Router instance. The most straightforward way for that is to move the construction of the server mixin into `data()`. `router` is useful for creating an instantsearch router compatible with vueRouter

### CODE REVIEW

**Review by eunjae-lee — APPROVED** [review]
makes sense to me

File: package.json [hunk 1]
```diff
@@ -103,6 +103,7 @@
     "vue-jest": "2.6.0",
     "vue-json-tree": "0.3.3",
     "vue-loader": "14.2.2",
+    "vue-router": "^3.4.3",
```
Comment (francoischalifour) [inline package.json:106]: Do we want to pin that dep?

Comment (Haroenv) [reply package.json:106]:
```suggestion
    "vue-router": "3.4.3",
```

File: yarn.lock [hunk 2]
```diff
@@ -15612,6 +15612,11 @@ vue-property-decorator@^8.0.0:
   dependencies:
     vue-class-component "^7.0.1"
 
+vue-router@^3.4.3:
```
Comment (Haroenv) [inline yarn.lock:15615]:
```suggestion
vue-router@3.4.3:
```

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**